mempool/priority/mempool.go

```diff
@@ -0,0 +1,819 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/config"
+	"github.com/cometbft/cometbft/libs/clist"
+	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cometbft/cometbft/libs/trace"
+	"github.com/cometbft/cometbft/mempool"
+	"github.com/cometbft/cometbft/proxy"
+	"github.com/cometbft/cometbft/types"
+)
+
+var _ mempool.Mempool = (*TxMempool)(nil)
+
+// TxMempoolOption sets an optional parameter on the TxMempool.
+type TxMempoolOption func(*TxMempool)
+
+// TxMempool implemements the Mempool interface and allows the application to
+// set priority values on transactions in the CheckTx response. When selecting
+// transactions to include in a block, higher-priority transactions are chosen
+// first.  When evicting transactions from the mempool for size constraints,
+// lower-priority transactions are evicted sooner.
+//
+// Within the mempool, transactions are ordered by time of arrival, and are
+// gossiped to the rest of the network based on that order (gossip order does
+// not take priority into account).
+type TxMempool struct {
+	// Immutable fields
+	logger       log.Logger
+	config       *config.MempoolConfig
+	proxyAppConn proxy.AppConnMempool
+	metrics      *mempool.Metrics
+	cache        mempool.TxCache // seen transactions
+
+	// Atomically-updated fields
+	txsBytes int64 // atomic: the total size of all transactions in the mempool, in bytes
+
+	// Synchronized fields, protected by mtx.
+	mtx                  *sync.RWMutex
+	notifiedTxsAvailable bool
+	txsAvailable         chan struct{} // one value sent per height when mempool is not empty
+	preCheckFn           mempool.PreCheckFunc
+	postCheckFn          mempool.PostCheckFunc
+	height               int64     // the latest height passed to Update
+	lastPurgeTime        time.Time // the last time we attempted to purge transactions via the TTL
+
+	txs        *clist.CList // valid transactions (passed CheckTx)
+	txByKey    map[types.TxKey]*clist.CElement
+	txBySender map[string]*clist.CElement // for sender != ""
+	evictedTxs mempool.TxCache            // for tracking evicted transactions
+
+	traceClient trace.Tracer
```

Comment:
[not blocking]
we're not acutally using this anywhere, but we're not too interested in tracing v1 I don't think so we're good to remove here

Reply:
removed fully.